pkg/snapstore/gcs_snapstore.go

```diff
@@ -103,20 +100,32 @@ func NewGCSSnapStore(config *brtypes.SnapstoreConfig) (*GCSSnapStore, error) {
 	return NewGCSSnapStoreFromClient(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, config.MinChunkSize, chunkDirSuffix, gcsClient), nil
 }
 
-func getGCSStorageAPIEndpoint(path string) (string, error) {
-	if _, err := os.Stat(path); err == nil {
-		data, err := os.ReadFile(path)
+func getGCSStorageAPIEndpoint(config *brtypes.SnapstoreConfig) (string, error) {
+	if gcsApplicationCredentialsPath, isSet := os.LookupEnv(getEnvPrefixString(config.IsSource) + envStoreCredentials); isSet {
+		storageAPIEndpointFilePath := path.Join(path.Dir(gcsApplicationCredentialsPath), fileNameStorageAPIEndpoint)
+		endpoint, err := os.ReadFile(storageAPIEndpointFilePath)
```

Comment:
In this current flow, we are mandating that the `storageAPIEndpoint` field should always be present in the secret, otherwise this errors out. But that shouldn't be the case when using with real GCS isn't it? 

Previously, we were checking if the file is existing or not using `os.Stat` but now we are directly looking to read the file which may not always be present. Can you let me know if this is intended.

Reply:
You're right @anveshreddy18 , I had missed this. Thanks for the find! Will push the fix.